### DISCUSSION TITLE
WiFi 7 won't connect

### DISCUSSION THREAD

**Discussion by TomInIowa** [espressif/arduino-esp32#10655] [Q&A]
None of my esp32-c6  chips will connect to my Asus RT-BE88U router when it is in WiFi  7 mode.   Is it a library problem?  hardware problem?     My new Arduino Uno R4 WiFi will NOT connect either. (it has a esp32-s3 chip)

**Comment by peterwillcn** [comment]
any news?

**Comment by lucasssvaz** [comment]
cc @me-no-dev

**Comment by RichardGladson** [comment]
# Wi-Fi 7 _doesn't work_ for microcontrollers
The ESP32-S3 chip uses __Wi-Fi 4__, while the ESP32 C6 uses __Wi-Fi 6__. Your router uses Wi-Fi 7, which is a newer protocol that makes it impossible to communicate with the chips. For it to work, you'll need to turn off Wi-Fi 7.

It is not a library problem or code problem, just a _protocol problem_.

**Comment by SuGlider** [comment]
Wi-Fi 7 should be fully backward compatible with Wi-Fi 6 (802.11ax) and older standards like Wi-Fi 4/5 on 2.4GHz.

But there is a known protocol and driver-level software compatibility bug between **older Broadcom-based Wi-Fi 7 routers** (like your **Asus RT-BE88U** and legacy client chips.

When you toggle your Asus RT-BE88U router into native "Wi-Fi 7 mode," it activates several advanced features that break the network handshake for non-Wi-Fi 7 IoT devices:

1. **MLO (Multi-Link Operation) Complications**: MLO allows Wi-Fi 7 devices to bind multiple bands (2.4, 5, and 6 GHz) into a single connection. Even though it is supposed to be backward compatible, early Broadcom Wi-Fi 7 router chipsets frequently drop or refuse connections from legacy devices when MLO is globally active.

2. **Aggressive RTS/CTS & OFDMA**: Wi-Fi 7 introduces new packet scheduling mechanisms. The current network library frameworks used by Arduino and older ESP-IDF versions do not handle these timing shifts well, causing a timeout during the initial DHCP/connection handshake

3. **Protected Management Frames (PMF) & WPA3**: Wi-Fi 7 mandates WPA3 and PMF. If your Arduino code or internal firmware isn't explicitly configured to negotiate PMF properly, the router will abruptly reject the device.

### How to Fix It (Ranked by Effectiveness)

1. **Turn Off MLO on the Router (Highly Recommended)**
ASUS officially acknowledges this issue with IoT devices. Disabling Multi-Link Operation fixes the handshake without crippling your overall Wi-Fi 7 speeds.
- Log into your ASUS router gateway dashboard (192.168.50.1 or router.asus.com).
- Navigate to Wireless > MLO.
- Turn Enable MLO to Disable

2. **Create an "IoT Network" or Guest Network SSID**
If you want to keep MLO active for your Wi-Fi 7 computers and phones, isolate your microcontrollers.
- In the ASUS dashboard, go to Guest Network Pro (or Network depending on your firmware).
- Create a dedicated IoT Network or standard 2.4 GHz Guest SSID.
- Inside the settings for that specific profile, make sure WiFi 7 mode is toggled OFF and set the security strictly to WPA2/WPA3 Personal. This keeps your main network in blazing-fast Wi-Fi 7 mode while giving your ESP32 chips a stable, legacy-friendly sandbox.

3. **Update the Arduino Uno R4 Wi-Fi Firmware**
The ESP32-S3 chip on the Uno R4 runs a separate firmware layer handled by Arduino. Early versions of this firmware fail completely on Wi-Fi 7 environments.
- Open the Arduino IDE.
- Go to Tools > WiFi101 / WiFiNINA Firmware Updater.
- Select your Uno R4 and flash the latest available firmware version to update its internal ESP32-S3 Wi-Fi stack.

4. Explicitly Enable WPA3/PMF in your Arduino Code
If your router forces WPA3/WPA2 mixed mode, your code needs to support PMF. For your ESP32-C6 and Uno R4 sketches, try initializing the Wi-Fi structure with explicit relaxed security permissions if you are using advanced ESP-IDF or updated Arduino cores:

``` cpp
// For standard Arduino connection, sometimes forcing a disconnect 
// and resetting the mode clears out legacy cache:
WiFi.disconnect(true);
WiFi.mode(WIFI_STA);
WiFi.begin(SECRET_SSID, SECRET_PASS);
```
